Description: Use Debian's llvm and gmock paths. Also removes unused librt and
 libboost-thread links.

--- a/cpp/ycm/tests/CMakeLists.txt
+++ b/cpp/ycm/tests/CMakeLists.txt
@@ -30,8 +30,7 @@
 
 if ( USE_SYSTEM_GMOCK )
   set( CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake;${CMAKE_MODULE_PATH}" )
-  find_package( GTest REQUIRED )
-  find_package( GMock REQUIRED )
+  add_subdirectory( /usr/src/gmock gmock )
 else()
   add_subdirectory( gmock )
   set( GTEST_INCLUDE_DIRS ${gtest_SOURCE_DIR} ${gtest_SOURCE_DIR}/include )
@@ -80,8 +79,8 @@
 target_link_libraries( ${PROJECT_NAME}
                        ${SERVER_LIB}
                        ${CLIENT_LIB}
-                       ${GTEST_LIBRARIES}
-                       ${GMOCK_LIBRARIES} )
+                       gmock
+                       )
 
 
 if ( NOT CMAKE_GENERATOR_IS_XCODE )
--- a/cpp/ycm/CMakeLists.txt
+++ b/cpp/ycm/CMakeLists.txt
@@ -143,7 +143,7 @@
 if ( PATH_TO_LLVM_ROOT )
   set( CLANG_INCLUDES_DIR "${PATH_TO_LLVM_ROOT}/include" )
 else()
-  set( CLANG_INCLUDES_DIR "${CMAKE_SOURCE_DIR}/llvm/include" )
+  file (GLOB CLANG_INCLUDES_DIR SYS_LLVM_INCLUDE_PATHS /usr/lib/llvm-3.*/include)
 endif()
 
 if ( NOT IS_ABSOLUTE "${CLANG_INCLUDES_DIR}" )
@@ -172,7 +172,7 @@
 endif()
 
 if ( USE_SYSTEM_BOOST )
-  find_package( Boost REQUIRED COMPONENTS python filesystem system regex thread )
+  find_package( Boost REQUIRED COMPONENTS python filesystem system regex )
 else()
   set( Boost_INCLUDE_DIR ${BoostParts_SOURCE_DIR} )
   set( Boost_LIBRARIES BoostParts )
@@ -236,7 +236,9 @@
     # On Debian-based systems, llvm installs into /usr/lib/llvm-x.y.
     file( GLOB SYS_LLVM_PATHS "/usr/lib/llvm*/lib" )
     # Need TEMP because find_library does not work with an option variable
-    find_library( TEMP clang
+    find_library( TEMP
+                  clang-3.6
+                  clang-3.7
                   PATHS
                   ${ENV_LIB_PATHS}
                   /usr/lib
@@ -266,10 +268,6 @@
   set( CMAKE_INSTALL_RPATH "${EXTRA_RPATH}:${CMAKE_INSTALL_RPATH}" )
 endif()
 
-# Needed on Linux machines, but not on Macs and OpenBSD
-if ( UNIX AND NOT ( APPLE OR SYSTEM_IS_OPENBSD ) )
-  set( EXTRA_LIBS rt )
-endif()
 
 #############################################################################
 
